### PR TITLE
STM32: Set correct crystal settings on Discovery boards

### PR DESCRIPTION
**Pull request by hierophect**
During the implementation of low power mode a couple of Discovery boards were mistakenly marked as having low power crystals, but actually only have DNP slots that require the user to install them. This resulted in these dev boards being unable to boot fully due to infinite loops on failure in some of the STM32 HAL functions. 

This PR sets the Discovery boards crystal settings based on their manual data, which should help with the boot problems. Unfortunately, I don't have most of the Disco boards on my desk at the moment so any help testing these changes would be appreciated.

### CODE REVIEW

File: ports/stm/boards/nucleo_f746zg/mpconfigboard.h [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #define FLASH_PAGE_SIZE             (0x4000)
 
 #define BOARD_OSC_DIV (8)
+#define BOARD_HAS_LOW_SPEED_CRYSTAL (0)
```
Comment (k0d) [inline ports/stm/boards/nucleo_f746zg/mpconfigboard.h:37]: The nucleo f746zg has a LSE crystal installed, but not a HSE.

File: ports/stm/boards/stm32f746g_discovery/mpconfigboard.h [hunk 2]
```diff
@@ -39,5 +39,7 @@
 #define BOARD_OSC_PLLN (400)
 #define BOARD_OSC_PLLQ (9)
 
+#define BOARD_HAS_LOW_SPEED_CRYSTAL (1)
```
Comment (k0d) [inline ports/stm/boards/stm32f746g_discovery/mpconfigboard.h:42]: This is correct

File: ports/stm/boards/nucleo_f746zg/mpconfigboard.h (hunk 1 above)
Comment (hierophect) [reply ports/stm/boards/nucleo_f746zg/mpconfigboard.h:37]: I think it just grabs the 8MHz crystal from the STlink, like most ST dev boards, which seems to be identical as far as clock settings go.

Comment (k0d) [reply ports/stm/boards/nucleo_f746zg/mpconfigboard.h:37]: That's true...but it does have a crystal installed. X2
source: https://www.st.com/en/evaluation-tools/nucleo-f746zg.html (and my eyes..as I look at the board)

**Comment by hierophect** [comment]
@k0d right now we're just looking for correct boot behavior. See here: https://github.com/adafruit/circuitpython/issues/2952

File: ports/stm/boards/nucleo_f746zg/mpconfigboard.h (hunk 1 above)
Comment (hierophect) [reply ports/stm/boards/nucleo_f746zg/mpconfigboard.h:37]: Yep! Didn't mean to contest your correction about the 32k crystal, I mixed it up with another board. Working on testing locally.

**Comment by k0d** [comment]
@Hierophect#2551 Not sure what you're doing with this PR, what's wrong with the LSE on H7/F7?

**Comment by hierophect** [comment]
@k0d @tannewt Unfortunately, it looks like I can't mess with the settings on the F7 or H7s without messing up their boot behavior. I thought it was just the Nucleos after my first round of testing but now I'm having issues on the H746 discovery as well. 

What these boards need is a full implementation of the low power domain for the H7 and F7. I can work on that this week but for now I've made a TODO comment for each of them so this can at least still be submitted as a hotfix for the Discovery F4.

**Comment by hierophect** [comment]
@k0d Right now, the new low power code in port.c has some caused problems with certain combinations of settings that cause boards to fail to boot if their oscillator settings don't match the actual board or the settings in `clocks.c`. Specifically, each board using an external LSE needs to have the backup domain configured properly or it won't exit the boot stage. I tried to make the system cleanly fall back to an internal oscillator if it fails, but my testing is showing it isn't working as it should be, at least not reliably. 

Unfortunately this does make this PR a bit anemic as-is... it does at least fix the Discovery F4, which is why I made it in the first place. But perhaps I should just rework it completely into a general low power settings fix across all boards.

**Pull request merged**